File: liemipeli/src/main/java/com/antero/liemipeli/domain/Maksa.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.antero.liemipeli.domain;

/**
 *
 * @author Antero Oikkonen
 */
public class Maksa implements Sisaelin {

    private Taika taika;

    public void setTaika(Taika taika) {
        this.taika = taika;
    }

    @Override
    public String getNimi() {
        return "maksa";
    }

    @Override
    public Taika getTaikavoima() {
        return taika;
    }

}

File: liemipeli-kotlin/src/main/kotlin/com/antero/liemipeli/domain/elaimet/Maksa.kt
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.antero.liemipeli.domain.elaimet

import com.antero.liemipeli.domain.Sisaelin
import com.antero.liemipeli.domain.taiat.SisaelinTaika
import com.antero.liemipeli.domain.taiat.Taika

/**
 *
 * @author Antero Oikkonen
 */
class Maksa : Sisaelin {

    override val taikavoima: Taika get() =
        SisaelinTaika("Maksa","taika joka lisää maksan tehoa")
    override val nimi: String get() = "Maksa"

}
